packages/sanity/src/structure/panes/document/timeline/utils.ts

```diff
@@ -0,0 +1,83 @@
+import {type Chunk, type ChunkType} from 'sanity'
+
+export type NonPublishChunk = Omit<Chunk, 'type'> & {
+  type: Exclude<ChunkType, 'publish'>
+  parentId?: string
+}
+
+export type PublishChunk = Omit<Chunk, 'type'> & {
+  type: 'publish'
+  children: string[]
+  collaborators: Set<string>
+}
+
+export const isNonPublishChunk = (chunk: Chunk): chunk is NonPublishChunk =>
+  chunk.type !== 'publish'
+
+export const isPublishChunk = (chunk: Chunk): chunk is PublishChunk => chunk.type === 'publish'
```

Comment:
Might need to add some comments / tagging these as `@internal`, since I'm not sure if these exports will show up in the automated docs :)

Reply:
They are not exported from the package, so I think they shouldn't be picked by the automated docs. There is a check that runs on build which validates that if something is exported from the packages it includes the supported tags.

Reply:
Ah this makes sense! Disregard my comment :)